File: Pascal/String/randomator.pas
{
 *   Copyright (C) 2015 NikitaProAndroid
}

program randomator;

var
  input, output: text;
  originalString, currentString: string;
  tempSymbol: char;
  counter: double;
  randomVal1, randomVal2, StringLength: uint64;

begin
  assign(input, 'input.txt');
  reset(input);
  assign(output, 'output.txt');
  rewrite(output);
  
  readln(input, originalString);
  counter := 0;
  StringLength := length(originalString);
  Randomize();
  currentString := originalString;
  
  repeat
  begin
    randomVal1 := random(StringLength);
    randomVal2 := random(StringLength);
    if randomVal1 < 1 Then Inc(randomVal1);
    if randomVal2 < 1 Then Inc(randomVal2);
    
    tempSymbol := currentString[randomVal1];
    currentString[randomVal1] := currentString[randomVal2];
    currentString[randomVal2] := tempSymbol;
    
    counter := counter + 1;
  end;
  until (currentString = originalString);
  
  if counter < 2 then begin
    writeln(output, 'You are lucky, selected positions for replacing symbols are same in a first attemp!');
    writeln(output, 'Random values both equals ', randomVal1);
  end;
  
  writeln(output, 'SUCCESSFULLY FINISHED! ATTEMPS: ', counter:0:0, ' SYMBOLS: ', StringLength);
  writeln(output, 'Final string: ', currentString);
  close(output);
end.

File: C++/String/randomator.cpp
/*
 *   Copyright (C) 2016 NikitaProAndroid
*/

#include <cstdlib>
#include <ctime>
#include <cstring>
#include <fstream>
#include <iostream>
#include <string>

using namespace std;
string mixString(unsigned int rand1, unsigned int rand2, string str) {
	char tmp = str[rand2];
	str[rand2] = str[rand1];
	str[rand1] = tmp;
	return str;
}

int main()
{
	unsigned long startTime = clock();

	ifstream input("input.txt");
	if (input.fail() || input.peek() == ifstream::traits_type::eof()) {
		cerr << "input.txt does not exist or is empty!";
		exit(EXIT_FAILURE);
	}
	ofstream output("output.txt");
	string originalString, currentString;
	unsigned int length;
	unsigned long int counter = 0;

	srand( (int) time(NULL) );

	getline(input, originalString);
	input.close();

	length = originalString.length();

	currentString = originalString;

	do {
		currentString = mixString(rand() % length, rand() % length, currentString);
		counter++;
	} while (strncmp(originalString.c_str(), currentString.c_str(), length) != 0);

	if (counter > 1)
	    output << "SUCCESSFULLY FINISHED! ATTEMPS: " << counter << " TIME(ms): " << (clock() - startTime);
	else
		output << "You are lucky, selected positions for replacing symbols are the same in first attemp!";
	output.close();
    return 0;
}
